Code/Tools/ProjectManager/Source/GemCatalog/GemItemDelegate.cpp

```diff
@@ -169,6 +173,82 @@ namespace O3DE::ProjectManager
         return QStyledItemDelegate::editorEvent(event, model, option, modelIndex);
     }
 
+    const GemModel* GetGemModel(const QAbstractItemModel* model)
+    {
+        const GemSortFilterProxyModel* proxyModel = qobject_cast<const GemSortFilterProxyModel*>(model);
+        if (proxyModel)
+        {
+            return proxyModel->GetSourceModel();
+        }
+        else
+        {
+            return qobject_cast<const GemModel*>(model);
+        }
+    }
+
+    QString GetGemNameList(const QVector<QModelIndex> modelIndices)
+    {
+        QString list;
+        for (int i = 0; i < modelIndices.size(); ++i)
+        {
+            if (!list.isEmpty())
+            {
+                if (i == modelIndices.size() - 1)
+                {
+                    list.append(" and ");
+                }
+                else
+                {
+                    list.append(", ");
+                }
+            }
+
+            list.append(GemModel::GetDisplayName(modelIndices[i]));
+        }
+
+        return list;
+    }
+
+    bool GemItemDelegate::helpEvent(QHelpEvent* event, QAbstractItemView* view, const QStyleOptionViewItem& option, const QModelIndex& index)
+    {
+        if (event->type() == QEvent::ToolTip)
+        {
+            QRect fullRect, itemRect, contentRect;
+            CalcRects(option, fullRect, itemRect, contentRect);
+            const QRect buttonRect = CalcButtonRect(contentRect);
```

Comment:
We shouldn't need to calculate this more than once. Can we just calc once and store this buttonRect then pass it into funcs that need it?

Reply:
option.rect is can be different for every item

Reply:
Ah yea you are right, this is only called conditionally do that makes sense.